src/interactivity/base/InteractivityFactory.cpp

```diff
@@ -375,6 +376,38 @@ using namespace Microsoft::Console::Interactivity;
     return status;
 }
 
+void InteractivityFactory::SetOwner(HWND owner) noexcept
+{
+    _owner.store(owner, std::memory_order_relaxed);
+
+    if (_pseudoConsoleWindowHwnd)
+    {
+        // DO NOT USE SetParent HERE!
+        //
+        // Calling SetParent on a window that is WS_VISIBLE will cause the OS to
+        // hide the window, make it a _child_ window, then call SW_SHOW on the
+        // window to re-show it. SW_SHOW, however, will cause the OS to also set
+        // that window as the _foreground_ window, which would result in the
+        // pty's hwnd stealing the foreground away from the owning terminal
+        // window. That's bad.
+        //
+        // SetWindowLongPtr seems to do the job of changing who the window owner
+        // is, without all the other side effects of reparenting the window.
+        // See #13066
+        ::SetWindowLongPtrW(_pseudoConsoleWindowHwnd, GWLP_HWNDPARENT, reinterpret_cast<LONG_PTR>(owner));
+    }
+}
+
+void InteractivityFactory::SetVisibility(const bool isVisible) noexcept
+{
+    if (_pseudoConsoleWindowHwnd && IsIconic(_pseudoConsoleWindowHwnd) != static_cast<BOOL>(isVisible))
+    {
+        _suppressVisibilityChange.store(true, std::memory_order_relaxed);
+        ShowWindow(_pseudoConsoleWindowHwnd, isVisible ? SW_SHOWNOACTIVATE : SW_MINIMIZE);

```

Comment:
Are we certain that this is processed in order on all versions of Windows we care about?

Alternatively... should we do the management of `suppress` in a custom window message?

Reply:
open question

Reply:
Ah, I overlooked this one. When you say in-order, what do you mean? If it's what I think it is, then yes, this will work: `ShowWindow` is synchronous by spec. There's an alternative. `ShowWindowAsync`.

Reply:
perfect. thanks.